core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -82,33 +85,95 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
    * @param data
    * @return Parameters set in examining data
    */
-  override def preValidationPrepare(data: Dataset[Row]): Option[SplitterSummary] = {
-
-    import data.sparkSession.implicits._
-
-    val labels = data.map(r => r.getDouble(0) -> 1L)
-    val labelCounts = labels.groupBy(labels.columns(0)).sum(labels.columns(1)).persist()
-    val (resKeep, resDrop) = estimate(labelCounts)
-    labelCounts.unpersist()
-    setLabels(resKeep, resDrop)
+  override def preValidationPrepare(data: Dataset[Row], labelColNameOpt: Option[String]): Option[SplitterSummary] = {
+    val labelColName = labelColNameOpt.getOrElse(data.columns(0))
+    val labelColIdx = data.columns.indexOf(labelColName)
+
+    if (Try(getLabelsToKeep).toOption.isEmpty) {
+      val labelCounts = data.groupBy(labelColName).count().persist()
+      val res = estimate(labelCounts)
+      labelCounts.unpersist()
+      setLabels(res.labelsKept.distinct, res.labelsDropped.distinct, res.labelsDroppedTotal)
+    }
 
-    summary = Option(DataCutterSummary(labelsKept = getLabelsToKeep, labelsDropped = getLabelsToDrop))
+    val newSummary = summary
+      .collect {
+        case dcs: DataCutterSummary if dcs.preparedDF.isDefined => dcs
+      }
+      .getOrElse {
+        val labelMetaArr = getLabelsFromMetadata(data)
+        val labelSet = getLabelsToKeep.toSet
+
+        // discount unseen label
+        val integralLabelSet = 0.until(labelMetaArr.length - 1)
+          .map(_.toDouble)
+          .toSet
```

Comment:
I dont think you can convert this to a set - it is a sequence where the name is matched to the value by the index

Reply:
you are right Leah. I am also just getting rid of it because it was my old way to prevent multiple executions of trimming, which we now handle more explicitly